### PR TITLE
Add description of .cm* files to FAQ

### PR DESCRIPTION
**Pull request by af**
As a newcomer I found references to these files in compiler/build errors quite cryptic. Not sure if this is the best place to document them but it'd be nice to have this information accessible and searchable.

### CODE REVIEW

File: src/pages/community/faq.md [hunk 1]
```diff
@@ -82,3 +82,18 @@ BuckleScript is optimized for performance across the whole stack. You can try sl
...
+
+OCaml outputs a few different files during compilation, depending on your build target (native/bytecode/JavaScript):
+
+.cmi: Compiled interface (.rei/mli) file
+.cmx: Compiled object file for native output (via ocamlopt)
+.cmo: Compiled object file for bytecode output
+.cmj: Compiled object file for web (via BuckleScript)
+.cma: Library file for bytecode output (equivalent to C's .a files)
+.cmxa: Library file for native output
+.cmt: Contains a "Typetree" – basically the AST with all type info
```
Comment (jaredly) [inline src/pages/community/faq.md]: oops I said that wrong. its `Typedtree`, and you can link it to https://github.com/ocaml/ocaml/blob/4.02/typing/typedtree.ml

File: src/pages/community/faq.md [hunk 2]
```diff
@@ -82,3 +82,18 @@ BuckleScript is optimized for performance across the whole stack. You can try sl
 - Adding a few infinite loops here and there.
 - Stuffing a JavaScript build tool in the pipeline.
 - Dragging in more dependencies for writing a hello world.
+
+### I'm seeing some weird .cm[ixja] files referenced in a compiler error. Where do these files come from?
+
+OCaml outputs a few different files during compilation, depending on your build target (native/bytecode/JavaScript):
+
+.cmi: Compiled interface (.rei/mli) file
```
Comment (jaredly) [inline src/pages/community/faq.md]: this won't render well - could you prefix each line with `- ` to make it a markdown list?

**Comment by jaredly** [comment]
Thanks for putting this together!

**Comment by af** [comment]
D'oh on the list formatting, thanks for catching that. And thanks to you, hcarty and bsansouci for all the info in Discord!

**Comment by chenglou** [comment]
About time! I've had https://github.com/facebook/reason/wiki/OCaml-Ecosystem-Extensions-List but it hasn't been very discoverable

**Comment by af** [comment]
@chenglou Ah! This is great, I hadn't seen that wiki page. Maybe we could combine that, and this PR's content into a new page in the reasonml.github.io guide? I can take a stab at it, just let me know the best place

**Comment by chenglou** [comment]
@af in the FAQ section would be nice! Then I'll kill the wiki page. It was intended for easy edit mostly

**Pull request merged**